lib/internal/fs/promises.js

```diff
@@ -614,11 +619,12 @@ async function write(handle, buffer, offsetOrOptions, length, position) {
     return { __proto__: null, bytesWritten: 0, buffer };
 
   let offset = offsetOrOptions;
-  if (isArrayBufferView(buffer)) {
+  const bufferView = isArrayBuffer(buffer) ? new Uint8Array(buffer) : buffer;
```

Comment:
Nit: this can be reworded into
```suggestion
  if (isArrayBuffer(buffer) {
    buffer = new Uint8Aray(buffer);
  }
```
No strong opinion, both options have their pros and cons.

Reply:
I used a new variable and avoided reassigning `buffer` in this function because `buffer` is included in the return value and I wanted to preserve the behavior of returning whatever the caller passed in.

Reply:
You are right, this function have to return the exact value of passed `buffer`.
Does that mean that [`fs.write()`](https://github.com/nodejs/node/pull/46490/files/00952f797ce91b59d110cb478637cb6dd1e0a8cf#diff-9b2c5bf36570be7a09ba2597abdf6d9bc2cc1e19866b6674b0cf275c1188dcadR824-R826) should do the same as here?

Reply:
You're right, I didn't pay careful enough attention to how `fs.write()` was eventually delivering `buffer` to its callback, I need to fix that one.